### ISSUE TITLE
Stop naming CSS files with version in name (sb-instagram-2-1.min.css)

### ISSUE THREAD

**Issue by jasomdotnet** [smashballoon/instagram-feed#52]
Now, this is the css path:

`/wp-content/plugins/instagram-feed/css/sb-instagram-2-1.min.css?ver=2.1`

when I want to move css file to the footer (external css files are render blocking), I need to rewrite my code after every plugin release (in order to register right css file for the footer). 

```
function dequeue_styles_sb_instagram_styles() {
	wp_deregister_style( 'sb_instagram_styles' );
}

add_action( 'wp_print_styles', 'dequeue_styles_sb_instagram_styles', 20 );

function move_styles_to_footer() {
	wp_enqueue_style( 'instagram', get_site_url( null, '/wp-content/plugins/instagram-feed/css/sb-instagram-2-1.min.css', 'https' ), array(), get_bloginfo( 'version' ) );
}

add_action( 'get_footer', 'move_styles_to_footer' );
```

After standard name it should looks like as following:

```
function move_styles_to_footer() {
	wp_enqueue_style( 'instagram', get_site_url( null, '/wp-content/plugins/instagram-feed/css/style.min.css', 'https' ), array(), get_bloginfo( 'version' ) );
}

add_action( 'get_footer', 'move_styles_to_footer' );
```

**Comment by craigschlegel** [comment]
Hey jasomdotnet,

We have been changing the name of our CSS and JS file names in recent updates due to a significant amount of users using a combination of page caching, browser caching, and removing of the version number in the query var at the end of the file name. This was resulting in sites breaking due to the large changes in the JS file and html in version 2.0.

After awhile, we might stop doing this however I have a solution that should work even better than what you've been doing. On the "Customize" tab, we have a setting to "Enqueue CSS file with shortcode". This will load the CSS file in the footer as well as conditionally load it only on pages that have the feed. 

Try removing your current script and using this setting instead.

**Comment by jasomdotnet** [comment]
>  users using a combination of page caching, browser caching, and removing of the version number in the query var at the end of the file name. This was resulting in sites breaking due to the large changes in the JS file and html in version 2.0.

I didn't get this part. Are you saying there was a problem with /style.css?ver=2.1 way of cache?

> On the "Customize" tab, we have a setting to "Enqueue CSS file with shortcode".

I appreciate this part. 

I also suggest you to create something like latest-style.min.css (or simple style.css) and locate it on fixed url (then don't forget about mentioning this in plugin docs, readme...). Latest version of Wordpress is available on wordpress.org/latest.zip

**Comment by jasomdotnet** [comment]
Another suggestion: add option to move dependency files (`js` and `css`) to the footer of the site. **External scripts are render blocking.** After this you may write in plug-in description "this plug-in doesn't negatively affect performance (better Google page-speed insights score)". Now it does (despite we use it because it's good).

**Comment by jasomdotnet** [comment]
Aha, there is such a setting "Check this box if you'd like to enqueue the JavaScript file for the plugin in the head instead of the footer." Great.

**Comment by craigschlegel** [comment]
I suppose you might not be aware but this code is also on the WordPress.org plugin repository:
https://wordpress.org/plugins/instagram-feed/

For this question:
_I didn't get this part. Are you saying there was a problem with /style.css?ver=2.1 way of cache?_
We have over 900,000 users in which a small number seem to have a combination of enabling aggressive browser caching and plugin or bit of PHP code that strips off the query var for all CSS and JS files. When we released the 2.0 update that contained significant changes to everything, several feeds and sites broke due to the JS and CSS files not updating. These users were submitting a lot of support requests and some were leaving negative reviews so we decided to rename the file with every release for awhile in order to prevent any more issues.
